This site contains course materials for SISG Module 17: Computational Pipeline for WGS Data, July 27-29, 2022. The official SISG web page can be found [here]() (requires login) -- **course evaluations and certificates of completion will be available via this page.**

- **Instructors:** Ken Rice, Laura Raffield, and Matthew Conomos
- **Zoom Link:** []
- **Slack Discussion:** []

## Course Format

#### Lectures
Course material will be presented through lectures given via Zoom. Slides for lectures are linked in the schedule below, and recordings of the lectures will be posted afterwards.

#### Exercises 
Many of the lectures will be followed with hands-on exercises. Students will be split into Zoom breakout rooms where they can work through the exercises together. 

To work through the exercises, log into [NHLBI BioData Catalyst powered by Seven Bridges](https://platform.sb.biodatacatalyst.nhlbi.nih.gov) with your username and password -- we will use this platform for all live demonstrations during the course.

All of the R code and data can also be downloaded from the [github repository](https://github.com/UW-GAC/SISG_2022) from which the site is built and run on your local machine. Download the complete workshop data and exercises: https://github.com/UW-GAC/SISG_2022/archive/master.zip


## Schedule

NOTE: All times are Pacific Daylight Time (GMT-07:00)

**Wednesday, July 27th**

| Time | Topic | Lecture | Exercises/Discussion |
| --- | --- | --- | --- |
| 11:30am-11:40am | Introduction | [Slides](), [Recording]() | |
| 11:40am-12:00pm | Using BioData Catalyst in this Module (?) | [Slides](), [Recording]() | |
| 12:00pm-1:00pm | Intro to Genomic Data Structure (?) | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 1:00pm-1:30pm | _Break_ | | | 
| 1:30pm-1:50pm | Cloud computing intro (?) | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 1:50pm-2:30pm | Association Tests Part I: Background | [Slides](), [Recording]() | |

**Thursday, July 28th**

| Time | Topic | Lecture | Exercises/Discussion |
| --- | --- | --- | --- |
| 8:00am-8:50am | Association Tests Part II: Single Variant Tests | [Slides](), [Recording]() | |
| 8:50am-9:45am | GENESIS for Association Tests | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 9:45am-10:00am | _Break_ | | | 
| 10:00am-11:00am | Population Structure and Relatedness Inference | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 11:00am-12:00pm | _Lunch_ | | | 
| 12:00pm-1:15pm | Mixed Model Association Testing | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 1:15pm-1:30pm | _Break_ | | |
| 1:30pm-2:30pm | Null model explorer? locus zoom app? plotting app? | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |

**Friday, July 29th**

| Time | Topic | Lecture | Exercises/Discussion |
| --- | --- | --- | --- |
| 8:00am-8:45am | Association Tests Part III: Multiple Variant Tests | [Slides](), [Recording]() | |
| 8:45am-9:45am | Variant Annotation for Association Testing | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 9:45am-10:00am | _Break_ | | | 
| 10:00am-11:00am | Annotation Explorer | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 11:00am-11:30am | Advanced Annotation Strategies for Association Testing | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 11:30am-12:30pm | _Lunch_ | | |
| 12:30pm-1:00pm | ??? | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 1:00pm-1:15pm | _Break_ | | |
| 1:15pm-2:00pm | Recent Findings from Sequencing Studies | [Slides](), [Recording]() | [Markdown](), [R code](), [Recording]() |
| 2:00pm-2:30pm | Open Q&A | | [Recording]() |


## R packages used

- [GENESIS](http://bioconductor.org/packages/release/bioc/html/GENESIS.html)
- [SeqArray](http://bioconductor.org/packages/release/bioc/html/SeqArray.html)
- [SeqVarTools](http://bioconductor.org/packages/release/bioc/html/SeqVarTools.html)
- [SNPRelate](http://bioconductor.org/packages/release/bioc/html/SNPRelate.html)
- [TopmedPipeline](https://github.com/UW-GAC/analysis_pipeline/tree/master/TopmedPipeline)
- [tidyverse](https://www.tidyverse.org)
- [GGally](https://cran.r-project.org/web/packages/GGally)


## Resources

NHLBI BioData Catalyst Powered by Seven Bridges 

- [Getting Started Guide for SISG21 Module 16](https://drive.google.com/file/d/1LMlzot2GXPKCRmLH6BtsOi9PxKAoohYT/view?usp=sharing)

If you are new to R, you might find the following material helpful:

- [Introduction to R](http://faculty.washington.edu/kenrice/rintro/indexSEA15.shtml) materials from SISG Module 3
- Graphics with [ggplot2](https://ggplot2.tidyverse.org/)
- Data manipulation with [dplyr](http://dplyr.tidyverse.org/)









### Markdown

Markdown is a lightweight and easy-to-use syntax for styling your writing. It includes conventions for

```markdown
Syntax highlighted code block

# Header 1
## Header 2
### Header 3

- Bulleted
- List

1. Numbered
2. List

**Bold** and _Italic_ and `Code` text

[Link](url) and ![Image](src)
```

For more details see [Basic writing and formatting syntax](https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).

### Jekyll Themes

Your Pages site will use the layout and styles from the Jekyll theme you have selected in your [repository settings](https://github.com/UW-GAC/SISG_2022/settings/pages). The name of this theme is saved in the Jekyll `_config.yml` configuration file.

### Support or Contact

Having trouble with Pages? Check out our [documentation](https://docs.github.com/categories/github-pages-basics/) or [contact support](https://support.github.com/contact) and weâ€™ll help you sort it out.
